sdk/core/core-client-rest/src/operationOptionHelpers.ts

```diff
@@ -10,6 +10,7 @@ import { OperationOptions, RequestParameters } from "./common";
  */
 export function operationOptionsToRequestParameters(options: OperationOptions): RequestParameters {
   return {
+    ...options,
```

Comment:
Is this needed? Looks that we already explicitly assign all values except `requestOptions` below. I like explicity

Reply:
Thanks for the comment, I was just thinking if it's a little wierd to have different ways of setting contentType by if contentType is application/json or not ? Not sure @joheredi will have different opinion on that ?

Reply:
My comment is only to the added `...options` line.  `OperationOptions` has four properties, three of which are already explicitly assigned.  Why do we need to explode `options` to put `requestOptions` on the returned object?  There's no `contentType` in `options`, unless there's something that I missed?
https://github.com/Azure/azure-sdk-for-js/blob/d44974c9b8da4b483348857440e06e64a213c4d5/sdk/core/core-client-rest/src/common.ts#L121-L141

Reply:
Get your point, the story is that in RequestParameters we have contentType defined, https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client-rest/src/common.ts#L47 which means rlc customers could set the contentType with options.contentType in any RLC libraries, and in Modular layer, we have the contentType generated in options when the default contentType is not application/json https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/test/modularIntegration/generated/azure/core/src/models/options.ts#L9, but if it's application/json, we will not generate such options as we trust the default behavior. so customer won't be able to set the contentType with options.contentType even if they write something like https://github.com/Azure/autorest.typescript/pull/2236/files/2a249b62144fb2b2dcc12577ce3bdc43a02c1ddd#diff-22b66627043891270064e93b7527b6d20345dbba0095396acb9173cf7b0dc285R181-R183 this way. I was just thinking the user experience is a little weird. But I think we should probably revisit this in a different issue ?

Reply:
Ah, then I guess the parameter type probably should be updated to the following if we expect `contentType`?

```ts
export function operationOptionsToRequestParameters(options: OperationOptions & { contentType?: string }): RequestParameters {
  return {
    contentType: options.contentType,
    ...
```

Reply:
That makes more sense, thank you for the insights :)

Reply:
Sorry my bad, I should've looked more at existing types. I think it's better to add the `contentType?: string;` to `OperationRequestOptions` `https://github.com/Azure/azure-sdk-for-js/blob/a2a2d038cfb6990e25eb696ec3ab1e39d60ec4e5/sdk/core/core-client-rest/src/common.ts#L146`, then we can get it from `options.requestOptions?.contentType`.

Reply:
I will just skip this part in this PR, as I am not sure if we want to add contentType to requestOptions, if yes, the current generated code probably has some issues and also not sure if it's align with HLC.

Reply:
issue created here https://github.com/Azure/autorest.typescript/issues/2268